src/components/common/ColonyActions/ActionDetailsPage/ActionDetailsPageFeed/ActionDetailsPageEvent/ActionEventData.tsx

```diff
@@ -14,20 +15,41 @@ const displayName = 'common.ColonyActions.ActionDetailsPage.ActionEventData';
 const ActionEventData = ({
   actionData: { createdAt, transactionHash, type },
   actionData,
+  motionMessageData,
   eventName,
-}: Pick<ActionsPageEventProps, 'actionData' | 'eventName'>) => {
+}: Pick<
+  ActionsPageEventProps,
+  'actionData' | 'eventName' | 'motionMessageData'
+>) => {
   const { colony } = useColonyContext();
+  const { user: motionMessageInitiatorUser } = useUserByNameOrAddress(
+    motionMessageData.initiatorAddress || '',
+  );
```

Comment:
We should really get rid of this hook, given there are actually no use cases for it. In majority of cases we just want to get user by their address, and the `UserProfile` component just wrongly seems to be allowing either address or name (in the Dapp it's only the name).
Here, we know we want to get user by its address, so there's no point in running a redundant username query. Please consider just using the `getUserByAddress` query directly

Reply:
This is actually correct.

In the dapp we could only fetch a user from a name due to limitations, but fetching it from an address is actually something I had introduced (work actually done by Olga) when we moved over to the CDapp.

I think it should still stand, unless you have a proper argument against it.